src/persist/src/location.rs

```diff
@@ -372,6 +386,60 @@ pub trait Consensus: std::fmt::Debug {
     async fn truncate(&self, key: &str, seqno: SeqNo) -> Result<usize, ExternalError>;
 }
 
+#[async_trait]
+impl<A: Consensus + Sync + Send + 'static> Consensus for Tasked<A> {
+    async fn head(&self, key: &str) -> Result<Option<VersionedData>, ExternalError> {
+        let backing = self.clone_backing();
+        let key = key.to_owned();
+        mz_ore::task::spawn(
+            || "persist::task::head",
+            async move { backing.head(&key).await },
+        )
+        .wait_and_assert_finished()
```

Comment:
how sure are we that wait_and_assert_finished is the right thing here (vs returning ExternalError)?

Reply:
Let me say what I was thinking and @guswynn may have opinions.

There are two possible cases for the backing task to fail:
- The task panics. I don't think we actually catch panics in tasks, but if we did this helper seems to handle it fine.
- The task is cancelled. We never cancel the task ourselves, so this should only happen when the runtime is being shut down externally. The helper cleverly yields back into the runtime before panicking, so if the runtime is shutting down we just won't be run instead of panicking.

I don't feel super strongly either way, but this seemed like a nice use of the new thing!

Reply:
Yeah, I think this is. the best option:
- We abort on panic so the panic resumption this won't matter except in tests like sqllogic test
- this avoids shutdown bugs like ben mentioned

Reply:
that said, I'm 100% in favor of whatever gets this out. seems like either is ~fine